src/encoder.rs

```diff
@@ -504,17 +504,33 @@ impl<T: Pixel> FrameInvariants<T> {
     let w_in_b = 2 * config.width.align_power_of_two_and_shift(3); // MiCols, ((width+7)/8)<<3 >> MI_SIZE_LOG2
     let h_in_b = 2 * config.height.align_power_of_two_and_shift(3); // MiRows, ((height+7)/8)<<3 >> MI_SIZE_LOG2
 
+    let frame_rate = (config.time_base.den / config.time_base.num) as usize;
+    let min_tile_cols = (config.width - 1) / 4096;
+    // The minimum number of tiles is determined based on the following requirements:
+    // - A tile cannot be wider than 4096 pixels
+    // - A tile cannot be larger than 4096x2304 pixels
+    // - The tile size * the frame rate * a temporal ratio cannot exceed a given fixed rate
+    //   corresponding to 4K60 resolution with a 1.1 ratio.
+    let min_tiles = 1 + min_tile_cols.max(
+      (config.width * config.height) / (4096 * 2304)).max(
+      (config.width * config.height * frame_rate) / (4096_f64 * 2176_f64 * 60_f64 * 1.1) as usize
```

Comment:
1.0?

Reply:
The maximum rate given here is based on a temporal ratio of 1.1, so that half of the equation does need to be scaled. The left-hand side is unscaled.

Reply:
Oh OK.